src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -480,13 +480,18 @@ private static MethodBodyStatement DeserializeIntoObjectProperty(JsonPropertySer
             // Reading a property value
             if (jsonPropertySerialization.ValueSerialization is not null)
             {
-                return new[]
+                List<MethodBodyStatement> statements = new List<MethodBodyStatement>
                 {
                     CreatePropertyNullCheckStatement(jsonPropertySerialization, jsonProperty, propertyVariables, shouldTreatEmptyStringAsNull),
-                    DeserializeValue(jsonPropertySerialization.ValueSerialization, jsonProperty.Value, out var value),
-                    Assign(propertyVariables[jsonPropertySerialization], value),
```

Comment:
Can we call `New.Instance(jsonPropertySerialization.SerializedType, value)` in the `DeserializeValue` and keep this method as is.

Reply:
I don't believe so because the assignment we call is conditional